HW29/src/main/java/org/example/models/ProductRepository.java

```diff
@@ -0,0 +1,30 @@
+package org.example.models;
+
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Component
+public class ProductRepository {
+
+    private final List<Product> productsInStock = new ArrayList<>();
+
+    public ProductRepository() {
```

Comment:
please use the custom initialization mechanism during bean creation, like`@PostContruct` .

https://docs.spring.io/spring-framework/reference/core/beans/annotation-config/postconstruct-and-predestroy-annotations.html